src/OpenTelemetry.Extensions.Serilog/OpenTelemetrySerilogSink.cs

```diff
@@ -0,0 +1,110 @@
+// <copyright file="OpenTelemetrySerilogSink.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Microsoft.Extensions.Logging;
+using Serilog.Core;
+using Serilog.Events;
+
+namespace OpenTelemetry.Logs
+{
+    internal sealed class OpenTelemetrySerilogSink : ILogEventSink, IDisposable
+    {
+        private readonly OpenTelemetryLoggerProvider openTelemetryLoggerProvider;
+        private readonly bool includeFormattedMessage;
+        private readonly LogEmitter logEmitter;
+        private readonly bool disposeProvider;
+
+        public OpenTelemetrySerilogSink(OpenTelemetryLoggerProvider openTelemetryLoggerProvider, bool disposeProvider)
```

Comment:
Going to mirror the comment here...

Would it make sense to just instantiate the extensions with an emitter instead of the provider? So like Serilog would look like

```csharp
Log.Logger = new LoggerConfiguration()
    .WriteTo.OpenTelemetry(openTelemetryLoggerProvider.CreateEmitter()) // <- Register OpenTelemetry Serilog sink
    .CreateLogger();
```

I don't have an issue with that. BUT...

* At the moment it isn't possible because `CreateEmitter` is `internal`.
* It forces users to manage the lifetime of `OpenTelemetryLoggerProvider`. At the moment it is kind of nice if you are only using serilog + a single openTelemetryLoggerProvider you can just transfer ownership to serilog and it will be managed/disposed naturally.

Reply:
> At the moment it is kind of nice if you are only using serilog + a single openTelemetryLoggerProvider you can just transfer ownership to serilog and it will be managed/disposed naturally.

Yea, I agree this seems like a nice feature.